src/methods/environment.js

```diff
@@ -127,6 +127,11 @@ export class EnvironmentImplementation {
 
     // 5. If IsPropertyReference(V) is true, then
     if (this.IsPropertyReference(realm, V)) {
+      if (base instanceof AbstractValue) {
```

Comment:
Shouldn't this check that we are in a pure function?

Reply:
The ToObjectPartial call already does that check, and if this can safely coerce to an object for any other reason, then I figured it probably should.